### PR TITLE
Profile token generation with more settings

### PR DESCRIPTION
**Pull request by AllentDan**

### CODE REVIEW

File: benchmark/profile_generation.py [hunk 1]
```diff
@@ -141,7 +148,155 @@ def main(model_path: str,
...
+    prompt_tokens: int
+    completion_tokens: int
+    throughput: float
+    memory: float
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Regression Test')
+    parser.add_argument('--model-path',
+                        nargs='+',
```
Comment (lvhan028) [inline benchmark/profile_generation.py:237]: One model is enough. There is no requirement for profiling more than one model

File: benchmark/profile_generation.py [hunk 2]
```diff
@@ -141,7 +148,153 @@ def main(model_path: str,
...
+                        default=[512, 512, 1024, 1024])
+    parser.add_argument('--tp', type=int, help='Tensor parallel', default=1)
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    args = parse_args()
+    results: List[ProfileResult] = []
+    for batch in args.concurrency:
```
Comment (lvhan028) [inline benchmark/profile_generation.py:265]: Let's add a progress bar by applying `tqdm`

File: benchmark/profile_generation.py [hunk 3]
```diff
@@ -141,7 +148,153 @@ def main(model_path: str,
...
+            time.sleep(5)  # wait a while for releasing GPU mem
+            memory = MemoryMonitor.terminate()
+            results.append(
+                ProfileResult(model_name=output[0][0],
+                              batch=batch,
+                              prompt_tokens=prompt_tokens,
+                              completion_tokens=completion_tokens,
+                              throughput=output[0][1],
+                              memory=memory))
+    with open('profile_generation.csv', 'w') as csvfile:
```
Comment (lvhan028) [inline benchmark/profile_generation.py:286]: Can we add another argument to represent `profile_generation.csv`?

File: benchmark/profile_generation.py [hunk 4]
```diff
@@ -141,7 +148,153 @@ def main(model_path: str,
...
+def parse_args():
+    parser = argparse.ArgumentParser(description='Regression Test')
+    parser.add_argument('--model-path',
+                        type=str,
+                        help='benchmark test model path')
+    parser.add_argument('--concurrency',
+                        nargs='+',
+                        type=int,
+                        help='how many requests launched concurrently',
+                        default=[1, 8, 32, 64])
```
Comment (lvhan028) [inline benchmark/profile_generation.py:243]: The default value of `max_batch_size` of turbomind is 32. May change the default value of `concurrency` to [1, 8, 16, 32]

File: benchmark/profile_generation.py [hunk 5]
```diff
@@ -141,7 +151,175 @@ def main(model_path: str,
...
+            memory = MemoryMonitor.terminate()
+            device_count = MemoryMonitor.device_count.value
+            results.append(
+                ProfileResult(model_name=output[0][0],
+                              batch=batch,
+                              prompt_tokens=prompt_tokens,
+                              completion_tokens=completion_tokens,
+                              throughput_per_proc=output[0][1],
+                              throughput_per_node=output[0][1] / args.tp *
+                              device_count,
```
Comment (lvhan028) [inline benchmark/profile_generation.py:304]: device_count?

File: benchmark/profile_generation.py [hunk 6]
```diff
@@ -141,7 +151,175 @@ def main(model_path: str,
...
+        for prompt_tokens, completion_tokens in tqdm(
+                zip(args.prompt_tokens, args.completion_tokens)):
+            MemoryMonitor.start()
+            from functools import partial
+            from multiprocessing import Pool
+            profile_target = partial(profile_throughput,
+                                     concurrency=batch,
+                                     input_seqlen=prompt_tokens,
+                                     output_seqlen=completion_tokens,
+                                     tp=args.tp)
```
Comment (lvhan028) [inline benchmark/profile_generation.py:292]: `args.tp` might not be the real tp, since user might set `--tp` when they use `deploy.py`

**Review by irexyc — APPROVED** [review]

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**